Lesson 4

SAV2018 opened the pull request:
Реализована загрузка и показ 20 последних фото c flickr.com (пункт меню "Feed"):
- добавлены библиотеки Retrofit и Glide
- изменён FeedPresenter
- изменены FeedFragment, FeedAdapter и MainActivity
- изменена разметка в fragment_feed и list_item
- добавлены классы для работы с API и класс модели возвращаемого сервером ответа (Feed), изменен класс модели FeedItem

SAV2018 committed:
initial commit

SAV2018 committed:
- FeedPresenter with simulation of loading random test data
- fragment_feed layout
- FeedFragment

SAV2018 committed:
- FeedPresenter with simulation of loading random test data
- fragment_feed layout
- FeedFragment

SAV2018 committed:
- FeedPresenter: loading 20 recent photos from flickr.com via retrofit

SAV2018 committed:
Loading 20 recent published photos from flickr.com via Retrofit (menu item "Feed").
- FeedPresenter
- FeedFragment
- fragment_layout
- list_item
- MainActivity
- data/model/Feed & FeedItem

rumpilstilstkin commented on app/src/main/java/ru/sav/cloudclient/data/model/Feed.java:
```diff
@@ -0,0 +1,83 @@
+        @SerializedName("modified")
+        @Expose
+        public String modified;
+
```
если имя параметра совпадает с именем поля то @SerializedName не нужен

rumpilstilstkin commented on app/src/main/java/ru/sav/cloudclient/presenter/feed/FeedPresenter.java:
```diff
@@ -0,0 +1,83 @@
+public class FeedPresenter extends BaseApiPresenter<Feed, FeedView> implements Subscriber<Feed> {
+    private static final String TAG = "FeedPresenter";
+
+
```
почистите лишние строки

rumpilstilstkin commented on app/src/main/java/ru/sav/cloudclient/view/MainActivity.java:
```diff
@@ -91,4 +93,21 @@ void drawConnectionState() { // показываем состояние соед
+        if (!result) {
+            Toast.makeText(this, R.string.msg_no_internet_connection,
+                    Toast.LENGTH_LONG).show();
+        }
```
можно выводить такие сообщения не через Toast а через Snackbar

SAV2018 committed:
Loading 20 recent published photos from flickr.com via Retrofit (menu item "Feed").
- FeedPresenter
- FeedFragment
- fragment_layout
- list_item
- MainActivity
- data/model/Feed & FeedItem
